compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BRecordType.java

```diff
@@ -77,25 +75,12 @@ public String toString() {
 
         if (tsymbol.name != null && (tsymbol.name.value.isEmpty() || tsymbol.name.value.startsWith(DOLLAR))) {
             // Try to print possible shape. But this may fail with self reference hence avoid .
-            int count = 0;
-            Map<BType, String> typeNames = new HashMap<>();
             StringBuilder sb = new StringBuilder();
             sb.append(RECORD).append(SPACE);
             sb.append(CLOSE_LEFT);
             for (BField field : fields) {
                 sb.append(SPACE);
-                int tag = field.type.tag;
-                if (tag < TypeTags.XML || tag == TypeTags.NIL || tag == TypeTags.ANY || tag == TypeTags.ANYDATA) {
-                    sb.append(field.type);
-                } else {
-                    if (typeNames.containsKey(field.type)) {
-                        sb.append(typeNames.get(field.type));
-                    } else {
-                        String typeName = DOLLAR + field.type.getKind().typeName() + count++;
-                        sb.append(typeName);
-                        typeNames.put(field.type, typeName);
-                    }
-                }
+                sb.append(field.type);
                 sb.append(SPACE).append(field.name)
```

Comment:
We can now merge the previous appends into this right?

Reply:
Fixed.